app/models/gem_info.rb

```diff
@@ -113,9 +113,9 @@ def compute_compact_index_info
   def requirements_and_dependencies
     group_by_columns = "number, platform, sha256, info_checksum, required_ruby_version, required_rubygems_version, versions.created_at"
 
-    dep_req_agg = "string_agg(dependencies.requirements, '@' ORDER BY rubygems_dependencies.name)"
+    dep_req_agg = "string_agg(dependencies.requirements, '@' ORDER BY rubygems_dependencies.name, dependencies.id)"
 
-    dep_name_agg = "string_agg(coalesce(rubygems_dependencies.name, '0'), ',' ORDER BY rubygems_dependencies.name) AS dep_name"
+    dep_name_agg = "string_agg(coalesce(rubygems_dependencies.name, '0'), ',' ORDER BY rubygems_dependencies.name, dependencies.id) AS dep_name"
```

Comment:
Something I noticed while reviewing this is that, since the string aggregation here happens on the dependency name itself, I don't think the extra ordering criteria would make any difference in this case?